utility_functions.R

```diff
@@ -0,0 +1,110 @@
+# This function provides a general-purpose way to make an API request to 
+# NMDC's runtime API. Note that this function will only return the first page 
+# of results. The function's input includes the name of the collection 
+# to access (e.g. biosample_set), the filter to be performed, the maximum page 
+# size, and a list of the fields to be retrieved. 
+# It returns the metadata as a dataframe.
+
+get_first_page_results <- function(collection, filter, max_page_size, fields) {
+  og_url <- paste0(
+      'https://api.microbiomedata.org/nmdcschema/', 
+      collection, '?&filter=', filter, '&max_page_size=', max_page_size, '&projection=', fields
+      )
+  
+  response <- jsonlite::fromJSON(URLencode(og_url, repeated = TRUE))
+  
+  return(response)
+}
+
+
+# The get_next_results function uses the get_first_page_results function, 
+# defined above, to retrieve the rest of the results from a call with multiple 
+# pages. It takes the same inputs as the get_first_page_results function above:
+# the name of the collection to be retrieved, the filter string, the maximum 
+# page size, and a list of the fields to be returned. This function returns the
+# results as a single dataframe (can be nested). It uses the next_page_token 
+# key in each page of results to retrieve the following page.
+
+get_next_results <- function(collection, filter_text, max_page_size, fields) {
```

Comment:
You might consider renaming this function to something like `get_all_results` because that's actually what it's doing (not just getting the next ones).